R/GeospatialFunctions.R

```diff
@@ -327,6 +385,55 @@ fetchATTAINS <- function(.data) {
     
     catchment_features <- feature_downloader(baseurls = baseurls[1], sf_bbox = bbox)
     
+    if(length(catchment_features) == 0 || is.null(catchment_features)){print("There are no ATTAINS features associated with your area of interest.")} else{
+    
+    ## GRABBING WATER TYPE:
+    
+    # Use ATTAINS API to grab, for each assessment unit, its WaterType.
+    # Query the API in "chunks" so it doesn't break:
+    split_vector <- function(vector, chunk_size = 2000) {
```

Comment:
suggestion: Could this function be defined outside of a conditional statement so that the definition only need to be included once?

Reply:
Good idea - will shift it to outside the conditional statement ahead of submitting to the EPA!